src/components/Provider.tsx

```diff
@@ -39,7 +39,7 @@ function Provider<A extends Action = AnyAction>({
     }
   }, [store, serverState])
 
-  const previousState = useMemo(() => store.getState(), [store])
+  const previousState = useMemo(store.getState, [store])
```

Comment:
It seems that it breaks custom stores that rely on `this`. A test covering this use case was originally introduced [here](https://github.com/reduxjs/react-redux/pull/465) and fixed [this issue](https://github.com/reduxjs/react-redux/issues/464)

Given how rare of a use case this is and the allocation cost that this requires for all users I would say that it would be better to drop support for this use case and require users who need this to bind their stores appropriately. It's an easy thing to do and feels to me like something that can be easily handled in the userland.

OTOH `Provider` itself is, at least currently, usually not rendered often as it lives "on the top" of the app. So kinda :shrug:

Reply:
Typically `Provider `is only ever going to render once in an app anyway. Given that, I'm not sure this change really provides any benefit :)